crates/stc_ts_type_checker/tests/conformance/types/intersection/intersectionOfUnionNarrowing.ts.stats.rust-debug

```diff
@@ -1,6 +1,6 @@
 Stats {
     required_error: 0,
     matched_error: 0,
-    extra_error: 0,
+    extra_error: 1,
```

Comment:
`else narrowing` is incorrect

```ts
declare const a : ((undefined | {
    bProp: string;
}) & ( object)); // { bProp: string; } & object
declare const b : ((undefined | {
    bProp: string;
}) & ( undefined | object)); // ({ bProp: string;} & object) | undefined
```
It should be evaluated as A, but it is evaluated as B